agent/app/data.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	"github.com/aws/amazon-ecs-agent/agent/data"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/eventstream"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	objNotFoundErrMsg = "not found"
+)
+
+// savedData holds all the data that we might save in db.
+type savedData struct {
+	taskEngine               engine.TaskEngine
+	agentVersion             string
+	availabilityZone         string
+	cluster                  string
+	containerInstanceARN     string
+	ec2InstanceID            string
+	latestTaskManifestSeqNum int64
+}
+
+// loadData loads data from previous checkpoint file, if any, with backward compatibility preserved. It first tries to
```

Comment:
thank you for the detailed explanation in the comment!

Reply:
no problem :)